bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -35,6 +35,52 @@ pub struct Room {
     timeline: RwLock<Option<Arc<Timeline>>>,
 }
 
+pub enum MembershipState {
+    /// The user is banned.
+    Ban,
+
+    /// The user has been invited.
+    Invite,
+
+    /// The user has joined.
+    Join,
+
+    /// The user has requested to join.
+    Knock,
+
+    /// The user has left.
+    Leave,
+}
+
+pub struct RoomMember {
+    pub user_id: String,
+    pub display_name: Option<String>,
+    pub avatar_url: Option<String>,
+    pub membership: MembershipState,
+    pub is_name_ambiguous: bool,
+    pub power_level: i64,
+    pub normalized_power_level: i64,
+}
+
+impl From<matrix_sdk::ruma::events::room::member::MembershipState> for MembershipState {
+    fn from(m: matrix_sdk::ruma::events::room::member::MembershipState) -> Self {
+        match m {
+            matrix_sdk::ruma::events::room::member::MembershipState::Ban => MembershipState::Ban,
+            matrix_sdk::ruma::events::room::member::MembershipState::Invite => {
+                MembershipState::Invite
+            }
+            matrix_sdk::ruma::events::room::member::MembershipState::Join => MembershipState::Join,
+            matrix_sdk::ruma::events::room::member::MembershipState::Knock => {
+                MembershipState::Knock
+            }
+            matrix_sdk::ruma::events::room::member::MembershipState::Leave => {
+                MembershipState::Leave
+            }
+            _ => unreachable!("unsupported MembershipState"),
```

Comment:
I think this deserves a TODO at least, this will crash the app if somebody sends unspecced membership state in an event.

Reply:
Makes sense to update as `todo!`?

Reply:
Doesn't really matter how we crash this, what matters is that we have a TODO comment that links to a github issue.